### PR TITLE
added osx plugin

### PR DESCRIPTION
**Pull request by CYKEB**
I created a plugin with integration for osx, based on a similar plugin for oh-my-zsh

### CODE REVIEW

File: plugins/osx/osx.load [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+#          FILE:  osx.fish
+#   DESCRIPTION:  oh-my-fish plugin file.
+#        AUTHOR:  Felix Sonntag
+#       VERSION:  1.0
+#       COMMENT:  Inspired by the oh-my-zsh plugin by Sorin Ionescu (sorin.ionescu@gmail.com)
+#                 https://github.com/robbyrussell/oh-my-zsh/blob/master/plugins/osx/osx.plugin.zsh
+# ------------------------------------------------------------------------------
+
+
+function tab
```
Comment (bpinto) [inline plugins/osx/osx.load]: Functions could receive a `-d` (description) argument so it's easier to understand what they do.
Each of the functions should live in their own file to make use of fish autoload functionality.

File: plugins/osx/osx.load [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  man -t $argv | open -f -a Preview
+end
+
+function trash
+  set -l trash_dir "$HOME/.Trash"
+  for item in $argv
+    if test -e $item
+      set -l item_name (basename $item)
+      if test -e "$trash_dir/$item_name"
+        set -l currentTime (date "+%H.%M.%S")
```
Comment (bpinto) [inline plugins/osx/osx.load]: Your code is snake cased, but `currentTime` is not.

File: plugins/osx/osx.load [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      else
+        mv -f "$item" "$trash_dir/"
+      end
+    end
+  end
+end
+
+function pfs
+  osascript 2>/dev/null -e '
+      set output to ""
```
Comment (bpinto) [inline plugins/osx/osx.load]: I don't understand this variable `output`, where is it used?

File: plugins/osx/osx.load [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        set output to output & ((item item_index of the_selection as alias)\'s POSIX path) & the_delimiter
+      end repeat'
+end
+
+
+function itunes
+  set -l opt $argv[1]
+  switch $opt
+    case launch play pause stop rewind resume quit
+      ;;
```
Comment (bpinto) [inline plugins/osx/osx.load]: Why do we need two `;`?

File: plugins/osx/osx.load [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        if item_index is item_count then set the_delimiter to ""
+        set output to output & ((item item_index of the_selection as alias)\'s POSIX path) & the_delimiter
+      end repeat'
+end
+
+
+function itunes
+  set -l opt $argv[1]
+  switch $opt
+    case launch play pause stop rewind resume quit
```
Comment (bpinto) [inline plugins/osx/osx.load]: Nothing necessary to send to itunes when one for any of these actions?

File: plugins/osx/osx.load [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      set opt "set mute to false"
+      ;;
+    case next previous
+      set opt "$opt track"
+      ;;
+    case vol volume
+      set opt "set sound volume to $argv[2]"
+      ;;
+    case "" -h --help
+      return 0
```
Comment (bpinto) [inline plugins/osx/osx.load]: Will this print the help below?

Comment (CYKEB) [inline plugins/osx/osx.load]: no, my fault

File: plugins/osx/osx.load (hunk 4 above)
Comment (CYKEB) [inline plugins/osx/osx.load]: I was not sure about the code style, at least in bash this is used as an explicit break. Gets fixed

File: plugins/osx/osx.load (hunk 5 above)
Comment (CYKEB) [inline plugins/osx/osx.load]: What do you mean with that?

Comment (bpinto) [inline plugins/osx/osx.load]: There is no `set opt "something"`

File: plugins/osx/osx.load (hunk 3 above)
Comment (CYKEB) [inline plugins/osx/osx.load]: line 66

Comment (bpinto) [inline plugins/osx/osx.load]: Isn't this variable always `""`?

`set output to ""`?

File: plugins/osx/osx.load (hunk 5 above)
Comment (CYKEB) [inline plugins/osx/osx.load]: `opt` is already set in line 72 and these commands don't take any additional options.

File: plugins/osx/osx.load (hunk 3 above)
Comment (CYKEB) [inline plugins/osx/osx.load]: That's AppleScript code, so you don't reference a variable with a `$`

File: plugins/osx/osx.load (hunk 4 above)
Comment (bucaran) [inline plugins/osx/osx.load]: There is no fall through inside `switch` statements in fish, so `break` is not required. 

`break` is only used inside loops, same as `continue`/

File: plugins/osx/osx.load (hunk 5 above)
Comment (bpinto) [inline plugins/osx/osx.load]: :+1:

File: plugins/osx/OSX-README.markdown [hunk 7]
```diff
@@ -0,0 +1,17 @@
+osx
```
Comment (bpinto) [inline plugins/osx/OSX-README.markdown]: this file needs to be named README only otherwise github won't parse it.

File: plugins/osx/cdf.fish [hunk 8]
```diff
@@ -0,0 +1,12 @@
+# cd to the current Finder directory
+
+function cdf
```
Comment (bpinto) [inline plugins/osx/cdf.fish]: `function cdf -d 'cd to the current Finder directory'`. This way fish shell will parse it and show it like this:

image

File: plugins/osx/itunes.fish [hunk 9]
```diff
@@ -0,0 +1,36 @@
+# Play, pause etc. iTunes
+
+function itunes
+  if count $argv >/dev/null
```
Comment (bpinto) [inline plugins/osx/itunes.fish]: `if [ (count $argv) -gt 0 ]`?

File: plugins/osx/manp.fish [hunk 10]
```diff
@@ -0,0 +1,12 @@
+# Open a specified man page in Preview
+
+function manp
+  if count $argv >/dev/null
+    switch $argv[1]
+      case -d --description
+        echo "Open a specified man page in Preview"
+        return 0
+    end
```
Comment (bpinto) [inline plugins/osx/manp.fish]: No else handling no arguments.

File: plugins/osx/pfd.fish [hunk 11]
```diff
@@ -0,0 +1,15 @@
+# Return the path of the frontmost Finder window
+
+function pfd
+  if count $argv >/dev/null
+    switch $argv[1]
+      case -d --description
+        echo "Return the path of the frontmost Finder window"
+        return 0
+    end
```
Comment (bpinto) [inline plugins/osx/pfd.fish]: No else handling no arguments.

File: plugins/osx/pfd.fish [hunk 12]
```diff
@@ -0,0 +1,15 @@
+# Return the path of the frontmost Finder window
+
+function pfd
+  if count $argv >/dev/null
+    switch $argv[1]
+      case -d --description
```
Comment (bpinto) [inline plugins/osx/pfd.fish]: This is probably not necessary if the function had the description defined on its declaration.

**Comment by CYKEB** [comment]
Thanks for all the feedback! I'm new to fish, so I'm sorry for the basic mistakes I have done.

**Comment by CYKEB** [comment]
That's odd that the tests failed on linux. The plugin shouldn't affect these tests.

**Comment by bpinto** [comment]
Don't worry about that, we all learn together :)

This test is randomly failing, but master has a possible fix, could you rebase your branch?

**Comment by bpinto** [comment]
The changes look good, after the rebase we can merge it.

I have a custom OSX plugin already, where I keep some functions I wrote, guess I'll have to rename it. :)

**Comment by CYKEB** [comment]
Great, thanks!
If your functions might be useful to other people you can extend the plugin :)

**Comment by bpinto** [comment]
Thank @CYKEB

**Comment by bpinto** [comment]
My functions were not osx related to be honest, I just didn't have a good name for it. :)
They are on my [dotfiles](https://github.com/bpinto/dotfiles), but they are not good enough to become plugins.

**Pull request merged**